src/praisonai/praisonai/api/call.py

```
@@ -351,32 +351,42 @@ def setup_public_url(port):
     if not OPENAI_API_KEY:
         raise ValueError('Missing the OpenAI API key. Please set it in the .env file or configure it through the GUI.')
-    
+
     if use_public:
         setup_public_url(port)
-    else:
-        print(f"Starting Praison AI Call Server on http://localhost:{port}")
-    uvicorn.run(app, host="0.0.0.0", port=port, log_level="warning")
+        host = "0.0.0.0"  # ngrok requires binding to all interfaces
+
+    if host == "0.0.0.0" and not use_public:
```

Ruff (0.15.12)

[error] Possible binding to all interfaces

(S104)